### PR TITLE
Release 2_2_4 patch

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Argument `has_hvac_flue` for ResidentialAirflow wasn't being assigned when no heating system was sampled.

The proposed solution here is to move the `has_hvac_flue` argument from the ResidentialAirflow measure out into the HVAC equipment measures (furnace, boiler, unit heater, shared systems). The value assigned to `has_hvac_flue` will be set as an additional property on the Building model object, and then the airflow measure will parse it from the Building model object. If it doesn't exist, then the value will be set to `false`.

## Checklist

Not all may apply:

- [ ] Unit tests have been added or updated
- [ ] All rake tasks have been run, and pass
- [ ] Documentation has been modified appropriately
- [ ] Any new options are added to `project_testing`
- [ ] `project_testing` runs without any failures
- [ ] No unexpected regression test changes
- [ ] All tests are passing (green) on circleci
- [ ] The [changelog](https://github.com/NREL/OpenStudio-BuildStock/blob/master/CHANGELOG.md) has been updated appropriately
- [ ] This branch is up-to-date with master

For more information on how to perform these checklist items, see the documentation's [Advanced Tutorial](https://resstock.readthedocs.io/en/latest/advanced_tutorial/index.html).

### CODE REVIEW

**Comment by aspeake** [comment]
In addition to `Heating Fuel ==` `None` and `Other Fuel`, I am seeing no HVAC systems for homes with all other heating fuels when the following parameters are assigned:
- `HVAC System Is Heat Pump == Yes `
- `HVAC System Is Shared == Neither`

There is also a small fraction of homes with `Heating Fuel == Electricity` and `HVAC System Is Heat Pump == No` that do not get an HVAC system assigned

**Comment by JLReyna** [comment]
@aspeake this is a mistake in the logic of the distributions (see #434 ). When we fix that issue so that homes assigned heating fuels are appropriately assigned heating systems, the proposed logic of the `has_hvac_flue` should work. I'm in the process of restructuring all of the HVAC TSVs, but it will probably be a month or so before I put in a PR. Are you wanting to see a quicker fix for the misassignment issue?

File: resources/measures/ResidentialHVACCentralSystemFanCoil/measure.rb [hunk 1]
```diff
@@ -62,6 +69,9 @@ def run(model, runner, user_arguments)
 
     fan_coil_heating = runner.getBoolArgumentValue("fan_coil_heating", user_arguments)
     central_boiler_fuel_type = HelperMethods.eplus_fuel_map(runner.getStringArgumentValue("central_boiler_fuel_type", user_arguments))
+    if fan_coil_heating
```
Comment (aspeake) [inline resources/measures/ResidentialHVACCentralSystemFanCoil/measure.rb:72]: Is this check needed? It looks like this logic is built into options_lookup, at least for the one `ResidentialHVACCentralSystemFanCoil` option with `fan_coil_heating=false` (`Fan Coil Cooling Only`), the flue gets set to false.

File: resources/options_lookup.tsv [hunk 2]
```diff
@@ -1222,275 +1222,275 @@ HVAC System Is Shared	Neither Shared
...
+		ResidentialAirflow																																													
 		ResidentialHVACSizing																																													
-HVAC System Heating Fuel Oil	Oil Boiler	ResidentialHVACBoiler	fuel_type=oil	"system_type=hot water, forced draft"	afue=0.76	oat_reset_enabled=false	oat_high=0.0	oat_low=0.0	oat_hwst_high=0.0	oat_hwst_low=0.0	design_temp=180.0	capacity=autosize	dse=NA																																		
-		ResidentialAirflow	has_hvac_flue=true																																												
+HVAC System Heating Fuel Oil	Oil Boiler	ResidentialHVACBoiler	fuel_type=oil	"system_type=hot water, forced draft"	afue=0.76	oat_reset_enabled=false	oat_high=0.0	oat_low=0.0	oat_hwst_high=0.0	oat_hwst_low=0.0	design_temp=180.0	capacity=autosize	dse=NA	has_hvac_flue=true																																	
+		ResidentialAirflow																																													
 		ResidentialHVACSizing																																													
 HVAC System Heating Wood	None																																														
 HVAC System Heating Wood	Wood Stove	ResidentialHVACUnitHeater	fuel_type=wood	efficiency=0.6	fan_power=0	airflow_rate=0	capacity=autosize
```
Comment (aspeake) [inline resources/options_lookup.tsv:1496]: Move flue argument from airflow to `ResidentialHVACUnitHeater`?

Comment (aspeake) [reply resources/options_lookup.tsv:1496]: ...This is meant to reference the `HVAC System Heating Wood` option

File: resources/measures/ResidentialHVACCentralSystemFanCoil/measure.rb (hunk 1 above)
Comment (joseph-robertson) [reply resources/measures/ResidentialHVACCentralSystemFanCoil/measure.rb:72]: My thought was that you could use the fan coil measure for cooling only. Then also have, e.g., the furnace measure for heating only. In such a case, you wouldn't want to overwrite the fan coil's `has_hvac_flue=false` with the furnace's `has_hvac_flue=true`.

File: resources/options_lookup.tsv (hunk 2 above)
Comment (joseph-robertson) [reply resources/options_lookup.tsv:1496]: Oops, missed one. Good catch.

File: resources/measures/ResidentialHVACCentralSystemFanCoil/measure.rb (hunk 1 above)
Comment (aspeake) [reply resources/measures/ResidentialHVACCentralSystemFanCoil/measure.rb:72]: Makes sense

**Comment by shorowit** [comment]
For consistency, the `has_water_heater_flue` argument could be moved to the appropriate water heater measures.

**Pull request merged**